fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rule/transformation/SplitAggregateRule.java

```diff
@@ -152,23 +152,51 @@ private void checkDistinctAgg(List<CallOperator> distinctAggCallOperator) {
         }
     }
 
-    private boolean hasAggregateEffect(OptExpression input) {
+    // True means good cases for two phase agg. It requires both group by keys and distinct cols are
+    // low cardinality which helps deduplication in the local stage.
+    private boolean hasAggregateEffect(OptExpression input, List<ColumnRefOperator> distinctColumns) {
         Statistics statistics = input.getGroupExpression().getGroup().getStatistics();
         Statistics inputStatistics = input.getGroupExpression().getInputs().get(0).getStatistics();
         Collection<ColumnStatistic> inputsColumnStatistics = inputStatistics.getColumnStatistics().values();
 
         if (inputsColumnStatistics.stream().anyMatch(ColumnStatistic::isUnknown)) {
             return false;
         }
-
         double inputRowCount = Math.max(1, inputStatistics.getOutputRowCount());
         double rowCount = Math.max(1, statistics.getOutputRowCount());
+
+        if (CollectionUtils.isNotEmpty(distinctColumns)) {
+            for (ColumnRefOperator distinctCol : distinctColumns) {
+                double ndv = inputStatistics.getColumnStatistic(distinctCol).getDistinctValuesCount();
+                rowCount = rowCount > ndv ? rowCount : ndv;
+            }
+        }
+
+
         if (rowCount * StatisticsEstimateCoefficient.LOW_AGGREGATE_EFFECT_COEFFICIENT < inputRowCount) {
             return true;
         }
         return false;
     }
 
+
+    // True means good cases for parallel deduplication. It requires group by keys have a relatively high
+    // cardinality (number of distinct value > 1000) to distribute data across multiple cores which helps
+    // parallel deduplication in the distinct global stage.
+    private boolean isSuitableForParallelization(OptExpression input) {
```

Comment:
not `number of distinct value > 1000`  which is `deduplication rate``?

`inputRowCount / StatisticsEstimateCoefficient.LOW_AGGREGATE_EFFECT_COEFFICIENT < rowCount` means outputRows/inputRowCount > 1/1000。

Reply:
Yes, just use the exist statistic info to evaluate the deduplication rate.

Reply:
the function name is confusing, suggest to use deduplication or parallelDeduplication

Reply:
why is this in reference to the concept `deduplication`?

Reply:
What about high cardinality but with data skew?

Reply:
If the cardinality is 128, and without skew, so data can be evenly shuffled. In this situation, this pr may lead to performance reduction?

Reply:
You're right. For scenes like high cardinality but still with data skew`, this pr help nothing. We may need to add other key to scatter the data and split the agg into more phases.
For the performace degradation problem, the new added cost exists two place. One is calculating hash value when exchange data, the other is split the top global agg into two phase agg. The first cost is really small according my test result. The second may lead some degradation when dataset is samller but may gain some profit when dataset is larger because the local streaming agg.